src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -116,6 +117,11 @@ function isNameEqual(x: string, y: string) {
 }
 
 export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
+    debouncedSetTags: any;
+    constructor(props: Readonly<ITagInputProps>) {
+        super(props);
+        this.debouncedSetTags = debounce(this.setTags, 5000);
```

Comment:
5 seconds might be too much. User could click to another page or close window in that time. Then, the project tags wont be saved for project. Have you tested shorter times?

Reply:
We need long enough time so It's not jittering on normal clicks. Yeah, maybe 5 sec little bit too much, I think about 3000ms should be enough. I just checked works fine, it hard to get on another page in less than 3sec.